### ISSUE TITLE
Color binding doesn't support alpha for hex strings missing an alpha portion

### ISSUE THREAD

**Issue by rcoopr** [cocopon/tweakpane#626]
When adding a binding with an initial value of a hex string with 3 or 6 digits (no alpha portion), the controller does not add an alpha slider.

```ts
const PARAMS = {
  tint: '#ff0',
};

pane.addBinding(PARAMS, 'tint', {
  view: 'color',
  color: { alpha: true },
});
```

I would expect declaring `{ alpha: true }` to always add a supporting alpha slider. As a workaround, you can add the alpha portion to the hex string.

| input | has alpha slider |
| ----------- | ----- |
| '#fff'                 | ❌          |
| '#ffffff'              | ❌          |
| '#fffc'                | ✅          |
| '#ffffffcc'            | ✅          |
| 0xfff                  | ✅          |
| 0xffffff               | ✅          |

**Comment by kitschpatrol** [comment]
I was initially confused about this as well, but  `{ alpha: true }` only applies to  how `number` values are parsed into colors, not `string` values.

(See [how params are processed for numbers](https://github.com/cocopon/tweakpane/blob/f3312dd5c61905611cce79d0bffa36b2e7340daa/packages/core/src/input-binding/color/plugin-number.ts#L65-L74) vs  [for strings](https://github.com/cocopon/tweakpane/blob/f3312dd5c61905611cce79d0bffa36b2e7340daa/packages/core/src/input-binding/color/plugin-string.ts#L50-L61).)

Similarly, the `{type: 'float' | 'int'}` param only applies to `object` color values.

This is consistent with how these parameters are [explained in the docs](https://tweakpane.github.io/docs/input-bindings/#color), but maybe it could be a bit clearer how certain parameters are limited to certain input value types.

**Comment by rcoopr** [comment]
I agree it's consistent, but it's very easy to think that the string input with alpha: true was just left out of the docs, because if you included every variation then nobody is going to read the 86 page long docs. 

I see that it's not a bug. I think it's not very useful/confusing to have a specific flag to include alpha but not have it work in some scenarios.

**Comment by kitschpatrol** [comment]
Yeah for sure — or some kind of generic typing approach that would restrict the available options based on the type of the bound value might help.

I'm working on a Tweakpane plugin to expand color handling by using the [ColorJS](https://colorjs.io/) library to back the internal color representation. This will bring support for CSS 4 color strings / more modern syntax, along with support for additional object / tuple param types. Hoping that can help with more demanding color use cases. Should be released in a week or two.
